### ISSUE TITLE
Feature Request - Filter to match each word separately

### ISSUE THREAD

**Issue by tonypdmtr** [sqlitebrowser/sqlitebrowser#532]
Suggestion:

When entering more than a word in a filter, instead of trying to match the phrase exactly with something like: 
COLUMN LIKE '%PHASE%'
it could match each word separately, like:
COLUMN LIKE '%WORD1%' AND COLUMN LIKE '%WORD2%' AND COLUMN LIKE '%WORD3%' etc.

This way it becomes quite a bit more flexible to search for records without having to remember the exact original phrase or which order the target words appear in the actual record.

**Comment by tonypdmtr** [comment]
I noticed in the latest release (don't know since when) that this feature is available but only for the "Filter in any column" box.

Is there a reason it is not available for the filter of each separate column?

**Comment by mgrojo** [comment]
You can get the same effect using a regular-expression filter. See in the [wiki](https://github.com/sqlitebrowser/sqlitebrowser/wiki/Using-the-Filters#regular-expression-filter).

In your case it would be:
`/^(?=.*\bWORD1\b)(?=.*\bWORD2\b).*$/`

Regular expressions aren't easy to parse nor to produce, but they are useful. Usually you only have to make a Google search to get the more complex cases.

**Comment by scottfurry** [comment]
I would recommend this [Online Regex Tester](https://regex101.com/r/gK0lX1/1). It allows to choose a _flavour_ of regex (JS, PHP, Python...), as well as showing the matches and capture group matches.

For the basics, I'll throw out [this gentle Intro](https://www.zytrax.com/tech/web/regex.htm).

**Comment by tonypdmtr** [comment]
> You can get the same effect using a regular-expression filter. 

If you're suggesting the use regular expressions as part of the implementation, it's a matter for those implementing it.  The end user wouldn't care about the internal implementation of a feature.
Although, I believe the SQL approach is still simpler to implement.

If however you're suggesting the end user to use regex, I think you'll drive all non-programmer users away.

**Comment by mgrojo** [comment]
No, I was suggesting it to you as a workaround for the lack of this feature. I did not close it because I consider it legitimate, but the least important factor is whether it is implemented as SQL or regexp. The question is how to change to behaviour from current one to what you suggest, because there would be users expecting current behaviour, which also makes sense. So some kind of option should be provided in the GUI.

**Comment by tonypdmtr** [comment]
But if the "filter in any column" already does that (and rightly so, AFAIC), isn't inconsistent to filter with different logic between the two?

**Comment by mgrojo** [comment]
There you have a point, but I'm reluctant to change current behaviour when it is not inherently wrong. Maybe the "filter in any column" should have worked the other way, but being "I don't have any idea where this is, so find it for me" has probably led to accept that design, which was also what one user asked for. I prefer the current behaviour, it is a simple containing filter, and also because it is aligned with what spreadsheets do (at least LibreOffice calc, but I guess Excel does the same). @MKleusberg implemented the global search, maybe he'd like to give his opinion.
